### ISSUE TITLE
[Configuration] Inject individual configurations

### ISSUE THREAD

**Issue by rinswind** [pkriens/cdi#25]
- **For**
    - Mashing up many configurations may become unusable
    - However the users may still need to depend on many configurations, but consume them separately
- **Suggestion**
    - When `@PID("org.example") @Configuration Map<String, Object> config;`
        - Inject that configuration only
        - Do not allow `@FactoryPID` on `@Configuration` injections
- **Note**: Must test how to make the synthetic beans work safely. I.e. it seems `@PID` must become a qualifier.

**Comment by rotty3000** [comment]
Isn't this exactly @pkriens' suggestion?
I don't think we ever intended @FactoryPID on injection points, only on @Component bean type.

**Comment by pkriens** [comment]
@FactoryPID is only on a component (it is the lifecycle PID)
We need @PID since you need to be able to specify required/optional. Configuration is always implied

**Comment by rinswind** [comment]
@rotty3000 yes it is the @pkriens suggestion. I added it here for discussion. Also I tend to agree this is needed. I added the explanationthat @FactoryPID is forbidden just in case of confusion. Somewhat my  own.

**Comment by rotty3000** [comment]
Can the initial description remove the qualifier comment?
I'd rather stick with `@PID` rather than put the argument on `@Configuration`!

**Comment by pkriens** [comment]
Yes, I like @PID, it makes it clear that there is a dependency which is absent for @Configuration and it allows multiple PIDs. I just wonder if @Configuration is necessary for that case? I'd like to restrict it to the application/component application only.

**Comment by rinswind** [comment]
Just `@PID` is not self-explanatory and we loose the symmetry with  `@Reference`
```
// Not very telling
@PID Map<String, ?> config; // Not very telling

// Configuration annotations don't have to have names that imply what are they for
@PID Connector config;
```

Also if we allow only `@PID` then we probably have to remove `@Configuration`. The user can use `@PID` with no value to signify they want the mashed up configuration.
```
@PID Config allTheConfig;
```

I propose we stick with `@PID @Configuration Config config;` because it is symmetric with `@Reference` where:
1. Use `@Reference` to signify the thing coming from OSGi
2. Use `@Filter` or property annotations to refine

**Comment by pkriens** [comment]
The PID requires a pid ...

     @PID("foo.bar.baz") Map<String,Object> map

Is imho very clear?

**Comment by rinswind** [comment]
Three options here
1. `@Inject @PID Map<String, Object> map;`
2. `@Inject @PID("org.example") @Configuration Map<String, Object> map;`
3. `@Inject @Configuration("org.example") Map<String, Object> map;`

(1.) and (2.) clearly relate the pid from `@Component` to the injection point.
To me (1.) looses clarity since `@PID` is a specialization of `@Configuration` rather than it's own thing.
(3.) Is compensates the clarity loss of (1.) but doesn't have the clear relation provided by (2.)

All in all I support (2.)

Also I am removing `action-required` since at present there is no clear decision.

**Comment by rotty3000** [comment]
I am in favour of 2.

**Comment by rinswind** [comment]
Then it is agreed
